source/solvers/vof_algebraic_interface_reinitialization.cc

```diff
@@ -0,0 +1,827 @@
+  const double n_dofs_per_cell =
+    fe_values_algebraic_reinitialization.get_fe().n_dofs_per_cell();
+
+  // Initialize local matrix
+  FullMatrix<double> local_matrix(n_dofs_per_cell, n_dofs_per_cell);
+
+  // Initialize local dof indices array
+  std::vector<types::global_dof_index> local_dof_indices(n_dofs_per_cell);
+
+  // Extractor for phase fraction gradient vector
+  FEValuesExtractors::Vector phase_fraction_gradients(0);
+
+  // Initialize phase fraction, projected phase gradient and projected curvature
+  // solution arrays
+  std::vector<double>         present_phase_fraction_values(n_q_points);
+  std::vector<Tensor<1, dim>> present_phase_gradient_projection_values(
+    n_q_points);
+  std::vector<double>         present_curvature_values(n_q_points);
+  std::vector<Tensor<1, dim>> present_reinitialized_phase_gradient_values(
+    n_q_points);
+
+  // Initialize shape function arrays
+  std::vector<double>         phi(n_dofs_per_cell);
+  std::vector<Tensor<1, dim>> grad_phi(n_dofs_per_cell);
+
+  // Compute diffusivity coefficient with the smallest cell size
+  const double h_min = identify_minimum_cell_size(
+    *this->mapping,
+    *this->subequations_interface->get_dof_handler(
+      VOFSubequationsID::algebraic_interface_reinitialization),
+    *this->cell_quadrature,
+    this->triangulation->get_communicator());
+  const double diffusivity_coefficient = compute_diffusivity(h_min);
+
+  // BDF coefficients for pseudo time-stepping
+  const Parameters::SimulationControl::TimeSteppingMethod method =
+    Parameters::SimulationControl::TimeSteppingMethod::bdf1;
+  const Vector<double> bdf_coefficient_vector =
+    calculate_bdf_coefficients(method, this->time_step_vector);
+
+  // Assemble system matrix
+  for (const auto &cell : this->dof_handler.active_cell_iterators())
+    {
+      if (cell->is_locally_owned())
+        {
+          // Reinitialize local matrix
+          local_matrix = 0;
+
+          // Get phase gradient projection and curvature cell iterators
+          typename DoFHandler<dim>::active_cell_iterator
+            vof_phase_gradient_projection_cell(
+              &(*this->triangulation),
+              cell->level(),
+              cell->index(),
+              dof_handler_vof_phase_fraction_gradient);
+          typename DoFHandler<dim>::active_cell_iterator
+            curvature_projection_cell(&(*this->triangulation),
+                                      cell->level(),
+                                      cell->index(),
+                                      dof_handler_curvature);
+
+          // Reinitialize FEValues with corresponding cell
+          fe_values_algebraic_reinitialization.reinit(cell);
+          fe_values_phase_gradient_projection.reinit(
+            vof_phase_gradient_projection_cell);
+          fe_values_curvature_projection.reinit(curvature_projection_cell);
+
+          // Get vector of Jacobi determinant times the quadrature weights
```

Comment:
```suggestion
          // Get vector of JxW 
```
That's the official deal.II terminology. Otherwise it would be the determinant of the jacobian * weight of the quadrature.